server/server.go

```diff
@@ -150,6 +150,7 @@ func (s *Server) Stop() {
 	log.Infof("Server endpoint will stop at [%v]", s.httpServer.Addr)
 	s.httpServer.Close()
 	observability.StopTracer(s.tracer)
+	s.conf.Close()
```

Comment:
If `conf` is managed outside of server then you may not want to close this here.

Reply:
I have not found a case where this would not be a good idea. If we find closing the config is an issue, we can take it out later, but it would require manual closing (likely just an issue for testing, I think).